src/core/account/account-files.js

```diff
@@ -51,43 +44,14 @@ function different(a: any, b: any): boolean {
   return false
 }
 
-/**
- * Returns `value` if it is an object,
- * otherwise returns an empty fallback object.
- */
-function getObject(value: any): any {
-  if (value == null && typeof value !== 'object') return {}
-  return value
-}
-
-function getJson(file: DiskletFile, fallback: any = {}): Promise<any> {
-  return file
-    .getText()
-    .then(text => JSON.parse(text))
-    .catch(e => fallback)
-}
-
-function getJsonFiles(folder: DiskletFolder): Promise<any[]> {
-  return mapFiles(folder, (file, name) =>
-    file
-      .getText()
-      .then(text => ({ file, name, json: JSON.parse(text) }))
-      .catch(e => undefined)
-  ).then(files => files.filter(file => file != null))
-}
-
 /**
  * Loads the legacy wallet list from the account folder.
  */
-async function loadWalletList(
-  folder: DiskletFolder
-): Promise<LoadedWalletList> {
-  const files = await getJsonFiles(folder.folder('Wallets'))
-
+async function loadWalletList(disklet: Disklet): Promise<LoadedWalletList> {
   const walletInfos: EdgeWalletInfo[] = []
   const walletStates = {}
-  for (const file of files) {
-    const clean = asMaybe(asLegacyWalletFile)(file.json)
+  for (const path of justFiles(await disklet.list('Wallets'))) {
+    const clean = await getFileJson(disklet, path, asLegacyWalletFile)
```

Comment:
Not that this one matters too much, but can't we use Promise.all here?

Reply:
Ok.